airflow/models/baseoperator.py

```diff
@@ -276,6 +276,7 @@ def partial(
     task_display_name: str | None | ArgNotSet = NOTSET,
     logger_name: str | None | ArgNotSet = NOTSET,
     allow_nested_operators: bool = True,
+    execute_on_success_callback_when_skipped: bool = True,
```

Comment:
If we want to keep compatibility, shouldn’t think be False instead?

Also since we’re not going to accept features on 2.x anyway, maybe we should just get rid of this argument entirely and always just run the callback?

Reply:
Today on_success_callback execute if task is skipped therefore it should be True for compatibility. 

Upon reconsideration, I'm not in favor of adding a parameter solely for compatibility. 
Logically, the `on_success_callback` shouldn't execute if the task is skipped, so this should be the default behavior.
 I agree with @ephraimbuddy that it constitutes a breaking change, albeit as a bugfix. 
Therefore, I'll mark this PR as a draft and plan to reopen it for Airflow 3.

Reply:
Cool, I thought the current behaviour is to never execute the callback. You plan sounds good to me.